transport/conn.go

```diff
@@ -252,6 +273,8 @@ func (h *connHandshaker) Close() {
 	h.closed = true
 	h.cv.Broadcast()
 	for conn := range h.workq {
+		// This does not do anything because conn.open is
+		// false and conn.Close is a no-op
```

Comment:
I don't think this comment is *necessarily* true.  But I need to review ... it may be that I've missed something here.  The intent is to be able to abort the handshaker via Close, and that should close any connections that might not be fully established yet.  This code was "recently" (in terms of release numbers, but not necessarily dates) refactored to better deal with stalled handshakes, and I may have messed this up.